### PR TITLE
CM-612: Task vs Thread in C#

### PR DESCRIPTION
**Pull request by marco-luzzara**

### CODE REVIEW

File: threads-csharp/TaskVsThread/TaskVsThread/CarBuilding.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    virtual protected Suspension BuildSuspension(int supportedKg) => new Suspension(supportedKg);
+
+    virtual protected Painting Paint(string color, int bodyArea) => new Painting(color, bodyArea);
+
+    virtual protected void Test(Body body, IEnumerable<Suspension> suspensions, Engine engine)
+    {
+        if (
+            suspensions.Sum(s => s.SupportedKg) <= body.Weight
+            || engine.Horsepower * 4 <= body.Weight
+        )
```
Comment (PhilBroderick) [inline threads-csharp/TaskVsThread/TaskVsThread/CarBuilding.cs:29]: The formatting looks off here.

**Review by PhilBroderick — CHANGES_REQUESTED** [review]
A couple of formatting changes - check the feedback doc.

File: threads-csharp/TaskVsThread/TaskVsThread/CarBuildingWithThread.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        bodyThread.Start();
+
+        Engine engine = null!;
+        var engineThread = new Thread(() =>
+        {
+            engine = carBuilding.BuildEngine(horsePower);
+        });
+        engineThread.Start();
+
+        ConcurrentBag<Suspension> suspensions = new ConcurrentBag<Suspension>();
```
Comment (PhilBroderick) [inline threads-csharp/TaskVsThread/TaskVsThread/CarBuildingWithThread.cs:25]: This should be `var`

**Review by PhilBroderick — CHANGES_REQUESTED** [review]
Just one more small change. I have changed it in the article so only needs changed in the PR.

File: threads-csharp/TaskVsThread/TaskVsThread/CarBuildingWithThread.cs (hunk 2 above)
Comment (marco-luzzara) [reply threads-csharp/TaskVsThread/TaskVsThread/CarBuildingWithThread.cs]: Oops, fixed, thanks.

File: threads-csharp/TaskVsThread/TaskVsThread/CarBuildingWithThread.cs [hunk 3]
```diff
@@ -22,7 +22,7 @@ public static void Build(int horsePower, int suspensionCount)
         });
         engineThread.Start();
 
-        ConcurrentBag<Suspension> suspensions = new ConcurrentBag<Suspension>();
+       var suspensions = new ConcurrentBag<Suspension>();
```
Comment (PhilBroderick) [inline threads-csharp/TaskVsThread/TaskVsThread/CarBuildingWithThread.cs:25]: This should have another space to be properly aligned, sorry super picky!

**Review by PhilBroderick — CHANGES_REQUESTED** [review]

File: threads-csharp/TaskVsThread/TaskVsThread/CarBuildingWithThread.cs (hunk 3 above)
Comment (marco-luzzara) [reply threads-csharp/TaskVsThread/TaskVsThread/CarBuildingWithThread.cs]: Sorry, I don't know how I messed up the indentation

**Review by PhilBroderick — APPROVED** [review]

**Pull request merged**